core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -2080,14 +2090,28 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
       maxConcurrent: Int
   )(f: O => F2[O2])(implicit F: Concurrent[F2]): Stream[F2, O2] = {
 
-    val init = ().pure[F2]
+    val init = F.unit
 
     def send(ch: Channel[F2, F2[Either[Throwable, O2]]], release: F2[Unit]) =
       (el: Either[Throwable, O2]) => release <* ch.send(el.pure[F2])
 
     parEvalMapAction(maxConcurrent, f)((ch, release) => init.as(send(ch, release)))
   }
 
+  /** Like parEvalMapUnordered but with unbounded concurrency.
+    */
+  def parEvalMapUnorderedUnbounded[F2[x] >: F[x], O2](
+      f: O => F2[O2]
+  )(implicit F: Concurrent[F2]): Stream[F2, O2] = {
+
+    val init = F.unit
+
+    def send(ch: Channel[F2, F2[Either[Throwable, O2]]], release: F2[Unit]) =
+      (el: Either[Throwable, O2]) => release <* ch.send(el.pure[F2])
+
+    parEvalMapUnboundedAction(f)((ch, release) => init.as(send(ch, release)))
```

Comment:
Is there any reason to use `init.as(----)` instead of `F.pure(send(ch, release))`?

Reply:
I don't know, this is just copy-paste of existing code. My best guess is some stack-safety trick.